ecosystem/python/sdk/aptos_sdk/account_address.py

```diff
@@ -34,10 +34,41 @@ def __eq__(self, other: object) -> bool:
         return self.address == other.address
 
     def __str__(self):
-        return self.hex()
+        """
+        Represent an account address in a way that is compliant with the v1 address
+        standard. The standard is defined as part of AIP-40, read more here:
+        https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-40.md
 
-    def hex(self) -> str:
-        return f"0x{self.address.hex()}"
+        In short, all special addresses SHOULD be represented in SHORT form, e.g.
+
+        0x1
+
+        All other addresses MUST be represented in LONG form, e.g.
+
+        0x002098630cfad4734812fa37dc18d9b8d59242feabe49259e26318d468a99584
+
+        For an explanation of what defines a "special" address, see `is_special`.
+
+        All string representations of addresses MUST be prefixed with 0x.
+        """
+        suffix = self.address.hex()
+        if self.is_special():
+            suffix = suffix.lstrip("0") or "0"
+        return f"0x{suffix}"
+
+    def is_special(self):
+        """
+        Returns whether the address is a "special" address. Addresses are considered
+        special if the first 63 characters of the hex string are zero. In other words,
+        an address is special if the first 31 bytes are zero and the last byte is
+        smaller than `0b10000` (16). In other words, special is defined as an address
+        that matches the following regex: `^0x0{63}[0-9a-f]$`. In short form this means
+        the addresses in the range from `0x0` to `0xf` (inclusive) are special.
+
+        For more details see the v1 address standard defined as part of AIP-40:
+        https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-40.md
+        """
+        return all(b == 0 for b in self.address[:-1]) and self.address[-1] < 0b10000
 
     @staticmethod
     def from_hex(address: str) -> AccountAddress:
```

Comment:
probably should relabel this `from_str` at some point

Reply:
I'll do it now while we're in the business of making breaking changes